gnupg-pkcs11-scd/command.c

```diff
@@ -1887,11 +1949,40 @@ gpg_error_t cmd_getattr (assuan_context_t ctx, char *line)
 	}
 	else if (!strcmp (l, "KEY-ATTR")) {
 		int i;
+		char buffer[1024];
+		const char *key_named_curve = NULL;
+		int keyAlgo;
+		int skip;
+
+		for (
+			pkcs11h_certificate_id_list_t curr_cert = user_certificates;
+			curr_cert != NULL;
+			curr_cert = curr_cert->next
+		) {
+			/* XXX:TODO: How do I know which key the KEY-ATTR is for ? */
+			error = get_cert_keyinfo(ctx, curr_cert->certificate_id, &keyinfo);
+			if (error != GPG_ERR_NO_ERROR) {
+				goto cleanup;
+			}
+
+		}
+
 		for (i=0;i<3;i++) {
-			char buffer[1024];
+			skip = 0;
+			switch (keyinfo_get_type(keyinfo)) {
+				case KEYINFO_KEY_TYPE_RSA:
```

Comment:
I would be happy if we can avoid any algo specific logic in `command.c`, this logic may also be  similar to the list.

Reply:
The format of message the SCD sends back to the agent differs based on the algorithm in a way that's specific to this interaction so it'll be a bit disjointed to put SCD command responses somewhere else.